### PR TITLE
PLAT-844: Skeleton component

### PR DESCRIPTION
**Pull request by f-jauregui**
⚠️  Unified the component to be flexible and easy to use. This is also an exploration so please feel free to comment if you think this approach is not good enough. ⚠️ 

[JIRA ticket](https://reciprocitylabs.atlassian.net/browse/PLAT-844)

#### Description
As per JIRA ticket. 

#### ChangeLOG

[Changelog](https://zen-ui.zengrc.com/?path=/docs/changelog--page)

#### Useful links
[Zen UI readme](https://github.com/reciprocity/zen-ui/blob/main/README.md)

[PR review guidelines](https://github.com/reciprocity/zengrc/blob/develop/doc/dev_environment_and_process/pull_request_reviews.md)

[Git commit guidelines](https://github.com/reciprocity/zengrc/blob/develop/doc/dev_environment_and_process/commit_guidelines.md)

### CODE REVIEW

**Review by homer0 — COMMENTED** [review]
LGTM :D; I'll leave the :shipit: when the tests are added.

File: src/components/zen-skeleton/zen-skeleton.scss [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+:host {
+  border-radius: 2px;
+  display: inline-block;
+  height: 100%;
+  width: 100%;
+  overflow: hidden;
+  position: relative;
+  background: $color-gray-100;
+
+  &::before {
```
Comment (chavlji) [inline src/components/zen-skeleton/zen-skeleton.scss:12]: Nice solution using pseudo!

Comment (chavlji) [inline src/components/zen-skeleton/zen-skeleton.scss:12]: Nice solution using pseudo!

File: src/components/zen-skeleton/zen-skeleton.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+})
+export class ZenSkeleton {
+  /** Width in any CSS width unit */
+  @Prop() readonly width: string = '100%';
+
+  /** Height in any CSS height unit */
+  @Prop() readonly height: string = '100%';
+
+  /** Makes the skeleton rounded */
+  @Prop({ reflect: true }) readonly isRounded: boolean = false;
```
Comment (chavlji) [inline src/components/zen-skeleton/zen-skeleton.tsx:16]: rename to `rounded`

File: src/components/zen-skeleton/zen-skeleton.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import { Component, Host, h, Prop } from '@stencil/core';
+
+@Component({
+  tag: 'zen-skeleton',
+  styleUrl: 'zen-skeleton.scss',
+  shadow: true,
+})
+export class ZenSkeleton {
+  /** Width in any CSS width unit */
+  @Prop() readonly width: string = '100%';
```
Comment (chavlji) [inline src/components/zen-skeleton/zen-skeleton.tsx:10]: I'd revemo this. Imo it's useless to duplicate suff that you can easily do with inline styling: `style="width: 100%; height:100%"`.

File: src/components/zen-skeleton/test/zen-skeleton.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  it('should render a squared skeleton', async () => {
+    const page = await newSpecPage({
+      components: [ZenSkeleton],
+      html: `<zen-skeleton width="100px" height="100px"></zen-skeleton>`,
+    });
+    expect(page.root.style.width).toBe('100px');
+    expect(page.root.style.height).toBe('100px');
+  });
+
+  it('should render a skeleton circle', async () => {
```
Comment (chavlji) [inline src/components/zen-skeleton/test/zen-skeleton.spec.tsx:22]: No need to test. Reflect is stencil's functionality.

**Review by chavlji — CHANGES_REQUESTED** [review]
Nice! Very clean and simple solution! Just a few minor tweaks...

**Review by chavlji — APPROVED** [review]

**Pull request merged**